### PR TITLE
Add extra convenience features to Snowflake

### PR DESCRIPTION
**Pull request by TellowKrinkle**
Some convenient features from the Sword project
Hopefully in the future, our conformance to RawRepresentable with UInt64 will get us an auto-implemented Comparable and Hashable, but for now it only gets us Equatable

### CODE REVIEW

File: Sources/SwiftDiscord/DiscordSnowflakeID.swift [hunk 1]
```diff
@@ -15,22 +15,27 @@
...
 public struct Snowflake {
 
-
     /// The internal ID storage for a snowflake
-    public let id: UInt64
+    public let rawValue: UInt64
+
+    /// For backwards compatibility, use rawValue instead
+    @available(*, deprecated, obsoleted: 4, renamed: "rawValue")
+    public var id: UInt64 { return rawValue }
```
Comment (nuclearace) [inline Sources/SwiftDiscord/DiscordSnowflakeID.swift]: No need for this. These change haven't been released yet.

File: Sources/SwiftDiscord/DiscordSnowflakeID.swift [hunk 2]
```diff
@@ -72,6 +77,41 @@ public typealias IntegrationID = Snowflake
 /// A Snowflake ID representing an Attachment
 public typealias AttachmentID = Snowflake
 
+// MARK: Extra snowflake information
+
+extension Snowflake {
```
Comment (nuclearace) [inline Sources/SwiftDiscord/DiscordSnowflakeID.swift:78]: Is this really needed? Is there a legit reason you would need this info?

**Review by nuclearace — CHANGES_REQUESTED** [review]

File: Sources/SwiftDiscord/DiscordSnowflakeID.swift (hunk 2 above)
Comment (TellowKrinkle) [reply Sources/SwiftDiscord/DiscordSnowflakeID.swift:78]: I can't think of any uses for the three `Int` values, but the Snowflake <-> Timestamp are pretty useful.  Should I remove `workerId`, `processId`, and `numberInProcess`?

**Review by nuclearace — APPROVED** [review]

File: Sources/SwiftDiscord/DiscordSnowflakeID.swift (hunk 2 above)
Comment (nuclearace) [reply Sources/SwiftDiscord/DiscordSnowflakeID.swift:78]: Nah it's fine. Maybe someone will find that info interesting.

**Pull request merged**